solidity/contracts/middleware/InterchainAccountRouter.sol

```diff
@@ -278,21 +281,41 @@ contract InterchainAccountRouter is Router {
         bytes32 _sender,
         bytes calldata _message
     ) external payable override onlyMailbox {
-        (
-            bytes32 _owner,
-            bytes32 _ism,
-            CallLib.Call[] memory _calls,
-            bytes32 _salt
-        ) = InterchainAccountMessage.decode(_message);
-
-        OwnableMulticall _interchainAccount = getDeployedInterchainAccount(
+        InterchainAccountMessage.MessageType _messageType = _message
+            .messageType();
+
+        // If the message is a reveal, we don't do any derivation of the ica address
+        // The commitment should be executed in the `verify` method of the CCIP read ISM that verified this message
+        // so here we just check that commitment -> ICA association has been cleared
+        if (_messageType == InterchainAccountMessage.MessageType.REVEAL) {
+            // The commitment should be executed in the `verify` method of the CCIP read ISM that verified this message
+            require(
+                verifiedCommitments[_message.commitment()] !=
+                    OwnableMulticall(payable(address(0))),
```

Comment:
don't you want to assert that it is equal to 0 (after the `delete` in `revealAndExecute`?